src/agent/src/image_rpc.rs

```diff
@@ -37,73 +38,58 @@ impl ImageService {
         Self { sandbox }
     }
 
-    fn build_oci_path(cid: &str) -> PathBuf {
-        let mut oci_path = PathBuf::from("/tmp");
-        oci_path.push(cid);
-        oci_path.push(IMAGE_OCI);
-        oci_path
-    }
-
-    fn pull_image_from_registry(image: &str, cid: &str, source_creds: &Option<&str>) -> Result<()> {
+    fn pull_image_from_registry(
+        image: &str,
+        cid: &str,
+        source_creds: &Option<&str>,
+        policy_path: &Option<&String>,
+    ) -> Result<()> {
         let source_image = format!("{}{}", "docker://", image);
 
-        let mut manifest_path = PathBuf::from("/tmp");
-        manifest_path.push(cid);
-        manifest_path.push("image_manifest");
-        let target_path_manifest = format!("dir://{}", manifest_path.to_string_lossy());
-
-        // Define the target transport and path for the OCI image, without signature
-        let oci_path = Self::build_oci_path(cid);
+        let tmp_cid_path = Path::new("/tmp/").join(cid);
+        let oci_path = tmp_cid_path.join(IMAGE_OCI);
         let target_path_oci = format!("oci://{}", oci_path.to_string_lossy());
 
-        fs::create_dir_all(&manifest_path)?;
         fs::create_dir_all(&oci_path)?;
 
         info!(sl!(), "Attempting to pull image {}...", &source_image);
 
         let mut pull_command = Command::new(SKOPEO_PATH);
         pull_command
-            // TODO: need to create a proper policy
-            .arg("--insecure-policy")
             .arg("copy")
             .arg(source_image)
-            .arg(&target_path_manifest);
+            .arg(&target_path_oci)
+            .arg("--remove-signatures");
```

Comment:
Please add a comment for this option explaining why it's required.

Reply:
Added a comment to reflect that umoci fails to unpack an image with signatures, so they need removing. In the original code there was a two phase copy, one to pull down and one to remove the signatures, but I combined them to save code, time and disk.